File: suites/13_simd_vectorization/comparisons/dot_product.cpp
#include <cstdio>

static long long dot_product(long long size) {
    long long sum = 0;
    for (long long i = 0; i < size; ++i) {
        long long a = ((i * 17 + 13) % 256) - 128;
        long long b = ((i * 19 + 7) % 256) - 128;
        sum += a * b;
    }
    return sum;
}

int main() { printf("%lld\n", dot_product(65536)); return 0; }

File: suites/13_simd_vectorization/comparisons/dot_product.rs
fn dot_product(size: i64) -> i64 {
    let mut sum: i64 = 0;
    for i in 0..size {
        let a = ((i * 17 + 13) % 256) - 128;
        let b = ((i * 19 + 7) % 256) - 128;
        sum += a * b;
    }
    sum
}

fn main() { println!("{}", dot_product(65536)); }
